db/migrate/20150407133716_create_tags.rb

```diff
@@ -0,0 +1,7 @@
+class CreateTags < ActiveRecord::Migration
+  def change
+    create_table :tags do |t|
+      t.string :name, null: false, unique: true
```

Comment:
You need an index to make a column unique. The `unique: true` option here is probably being ignored.